airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -0,0 +1,180 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class AwsGlueJobHook(AwsBaseHook):
+    """
+    Interact with AWS Glue - create job, trigger, crawler
+
+    :param job_name: unique job name per AWS account
+    :type job_name: Optional[str]
+    :param desc: job description
+    :type desc: Optional[str]
+    :param concurrent_run_limit: The maximum number of concurrent runs allowed for a job
+    :type concurrent_run_limit: int
+    :param script_location: path to etl script on s3
+    :type script_location: Optional[str]
+    :param retry_limit: Maximum number of times to retry this job if it fails
+    :type retry_limit: int
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Job
+    :type num_of_dpus: int
+    :param region_name: aws region name (example: us-east-1)
+    :type region_name: Optional[str]
+    :param iam_role_name: AWS IAM Role for Glue Job
+    :type iam_role_name: Optional[str]
+    :param s3_bucket: S3 bucket where logs and local etl script will be uploaded
+    :type s3_bucket: Optional[str]
+    """
+    JOB_POLL_INTERVAL = 6  # polls job status after every JOB_POLL_INTERVAL seconds
+
+    def __init__(self,
+                 s3_bucket=None,
+                 job_name=None,
+                 desc=None,
+                 concurrent_run_limit=1,
+                 script_location=None,
+                 retry_limit=0,
+                 num_of_dpus=10,
+                 region_name=None,
+                 iam_role_name=None,
+                 *args, **kwargs):
+        self.job_name = job_name
+        self.desc = desc
+        self.concurrent_run_limit = concurrent_run_limit
+        self.script_location = script_location
+        self.retry_limit = retry_limit
+        self.num_of_dpus = num_of_dpus
+        self.region_name = region_name
+        self.s3_bucket = s3_bucket
+        self.role_name = iam_role_name
+        self.s3_glue_logs = 'logs/glue-logs/'
+        super(AwsGlueJobHook, self).__init__(client_type='glue', *args, **kwargs)
+
+    def get_conn(self):
+        """
+        :return: connection
+        """
+        conn = self.get_client_type('glue', self.region_name)
+        return conn
```

Comment:
You don't need this anymore due to https://github.com/apache/airflow/pull/7541

Reply:
@feluelle Okay, i will push it with other changes(about the test that is failing). currently trying to figure it out.